extensions/amp-consent/integrating-consent.md

```diff
@@ -0,0 +1,88 @@
+A remote endpoint is expected to tell the AMP runtime whether the user consent is required. It can also pass extra information via AMP to 3P vendors. More information on the remote endpoint can be found [here]((https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/amp-consent.md#checkconsenthref)).
+
+## Prompt UI Iframe
+The AMP runtime will embed the CMP's prompt in an iframe. amp-consent will create the iframe only when it is necessary. A default placeholder will be displayed, and the prompt iframe will remain hidden until it has finished loading.
+
+The prompt iframe and the parent AMP page will communicate through postMessages and the iframe name attribute. In the case of using postMessage, messages from nested iframes will be ignored. The lists of support APIs are:
+
+#### Requesting UI state change
+Iframes can send a `consent-ui` message to the parent AMP page to request UI state change.
+
+``` javascript
+window.parent.postMessage({
+  type: 'consent-ui',
+  action: 'ready/enter-fullscreen'
+}, '*');
+```
+
+Action `'ready'` informs the AMP runtime to hide the placeholder and show the consent prompt instead.
+
+Action `'enter-fullscreen'` requests the AMP runtime to expand the iframe to fullscreen.
+
+#### Informing Consent response
+Iframes can send a `consent-response` message to the parent AMP page to inform the user [actions](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/amp-consent.md#prompt-actions) along with additional consent information.
+
+``` javascript
+window.parent.postMessage({
+  type: 'consent-response',
+  action: 'accept/reject/dismiss',
+  info: /string/
+}, '*');
+```
+
+* The user action is always required by AMP runtime. It allows the user to block/unblock any AMP component by user consent, not only components that read the additional consent information string.
+
+* If the user action value equals to `dismiss`, the `info` value should not be defined. This is because the `info` string is not useful when the user decides to not give any information.
+
+* If the user action value equals to `accept` or `reject`. AMP will strictly respect the incoming `info` value. If the `info` value is undefined, any previously value will be discarded.
+
+#### Client information passed to iframe
+When the iframe is created, the following information will be passed to the iframe via the name attribute.
+* `clientConfig`: The configuration from the publisher
+* `consentState`: The stored consent state if there's any. The value will be `'accepted'/'rejected'/'unknown'`. A friendly reminder is to be aware of the difference between the consent state string value (`'accepted'/'rejected'/'unknown'`) and the user action string value (`'accept'/'reject'/'dismiss'`).
+* `consentString`: The stored consent info string if there's any.
+
+The name attribute value will equal to
+``` javascript
+  JSON.stringify({
+    'clientConfig': {*},
+    'consentState': {string=},
```

Comment:
I am not sure what you mean by this code example, perhaps we could re-structure this? Or add more context with something like a comment that says `// inside the iframe`?